compiler/core/src/swift/swiftComponent.re

```diff
@@ -451,6 +451,21 @@ let generate =
       |> List.map(defineInitialLayerValues)
       |> List.concat;
     };
+    /* contains & isPropertyUsed should be made more general if needed in other places */
+    let contains = (~value: 'a, theList: list('a)) => {
+      let f = (found, elem) => found || elem == value;
+      ListLabels.fold_left(~f, ~init=false, theList);
+    };
+    let isPropertyUsed = (root, property) => {
+      let bindings = 
+        root
```

Comment:
If I'm understanding the child/parent border width issue you're mentioning correctly, I think you need to make sure that `isPropertyUsed` is checking just the specific layer it's called with, rather than the entire layer hierarchy. You'll also want to detect if one of these parameters is assigned by logic.

I tried it out to double check and I think this should be what you want:

```reason
    let isPropertyUsed = (layer: Types.layer, property) => {
      let assignedParameters =
        Layer.LayerMap.find_opt(layer, layerParameterAssignments);
      let parameterIsAssigned =
        switch assignedParameters {
        | Some(parameters) => StringMap.mem(property, parameters)
        | None => false
        };
      parameterIsAssigned || StringMap.mem(property, layer.parameters);
    };
```

Side note: probably can use `List.find` or `List.exists` for `contains`.

Reply:
Nice, I will give this a try shortly!